libbeat/management/management.go

```diff
@@ -33,27 +58,31 @@ var DebugK = "centralmgmt"
 
 var centralMgmtKey = "x-pack-cm"
 
-// ConfigManager interacts with the beat to update configurations
-// from an external source
-type ConfigManager interface {
-	// Enabled returns true if config manager is enabled
+// Manager interacts with the beat to provide status updates and to receive
+// configurations.
+type Manager interface {
+	// Enabled returns true if manager is enabled.
 	Enabled() bool
 
-	// Start the config manager
-	Start(func())
+	// Start the config manager giving it a stopFunc callback
+	// so the beat can be told when to stop.
+	Start(stopFunc func())
 
-	// Stop the config manager
+	// Stop the config manager.
 	Stop()
 
-	// CheckRawConfig check settings are correct before launching the beat
+	// CheckRawConfig check settings are correct before launching the beat.
 	CheckRawConfig(cfg *common.Config) error
+
+	// UpdateStatus called when the status of the beat has changed.
+	UpdateStatus(status Status, msg string)
```

Comment:
How about moving the UpdateStatus method into it's own interface? I'd like to push only 'status' reporting to the inputs/outputs and provide a set of helper functions that wrap a 'StatusReporter', merging status changes from multiple subsystems. But in the end we can also introduce the minimal interface in another package when we need it.

Reply:
I can move it to its own interface, would it also still be in the `management.Manager` interface?

```
type Manager interface {
    StatusReporter

    ...
}
```